### PR TITLE
Add ConstCheckedPow trait

### PR DESCRIPTION
**Pull request by kaidokert**
More const #58

## Summary by Sourcery

Introduce a const-compatible checked exponentiation capability and apply it to primitive integers and FixedUInt types.

New Features:
- Add the ConstCheckedPow trait for const-compatible checked exponentiation with overflow detection.
- Implement ConstCheckedPow for unsigned primitive integer types (u8, u16, u32, u64, u128).
- Provide a ConstCheckedPow implementation for FixedUInt using exponentiation by squaring with checked multiplication.
- Expose a const function wrapper const_checked_pow for FixedUInt to enable compile-time checked exponentiation.

Tests:
- Add runtime and const-evaluated tests verifying FixedUInt checked_pow behavior, including edge, overflow, and just-fitting cases.

## Summary by CodeRabbit

* **New Features**
  * Added const-compatible checked exponentiation: public checked_pow API for unsigned primitives (u8–u128) and fixed-size big integers, returning None on overflow and usable in const contexts.

* **Tests**
  * Added unit and const-evaluation tests covering edge cases, overflow scenarios, and a nightly-guarded compile-time evaluation path.

### CODE REVIEW

File: src/fixeduint/checked_pow_impl.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            let mut result = Self::one();
+            let mut base = self;
+            let mut e = exp;
+            while e > 0 {
+                if (e & 1) == 1 {
+                    result = ConstCheckedMul::checked_mul(&result, &base)?;
+                }
+                e >>= 1;
+                if e > 0 {
+                    base = ConstCheckedMul::checked_mul(&base, &base)?;
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/checked_pow_impl.rs]: this would require [feature(const_try)] - not enabled here
